eclair-core/src/test/scala/fr/acinq/eclair/router/ChannelRangeQueriesSpec.scala

```diff
@@ -127,4 +129,58 @@ class ChannelRangeQueriesSpec extends FunSuite {
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, false) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2))
     assert(Router.computeFlag(channels)(ef.shortChannelId, None, None, true) === (INCLUDE_CHANNEL_ANNOUNCEMENT | INCLUDE_CHANNEL_UPDATE_1 | INCLUDE_CHANNEL_UPDATE_2 | INCLUDE_NODE_ANNOUNCEMENT_1 | INCLUDE_NODE_ANNOUNCEMENT_2))
   }
+
+  def makeShortChannelIds(height: Int, count: Int): List[ShortChannelId] = (0 until count).map(c => ShortChannelId(height, c, 0)).toList
+
+  test("split short channel ids correctly (basic tests") {
+    assert(Router.split(SortedSet.empty[ShortChannelId], 100) == ShortChannelIdsChunk(0, 0, Nil) :: Nil)
+    assert(Router.split(SortedSet(ShortChannelId(1000, 0, 0)), 100) == ShortChannelIdsChunk(1000, 1, ShortChannelId(1000, 0, 0) :: Nil) :: Nil)
+
+    // all ids in different blocks, but they all fit in a single chunk
+    {
+      val ids = List(ShortChannelId(1000, 0, 0), ShortChannelId(1001, 0, 0), ShortChannelId(1002, 0, 0), ShortChannelId(1003, 0, 0), ShortChannelId(1004, 0, 0), ShortChannelId(1005, 0, 0))
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, ids.size)
+      assert(chunks == ShortChannelIdsChunk(1000, ids.size, ids) :: Nil)
+    }
+
+    // all ids in different blocks, chunk size == 2
+    {
+      val ids = List(ShortChannelId(1000, 0, 0), ShortChannelId(1001, 0, 0), ShortChannelId(1002, 0, 0), ShortChannelId(1003, 0, 0), ShortChannelId(1004, 0, 0), ShortChannelId(1005, 0, 0))
+      val chunks = Router.split(SortedSet.empty[ShortChannelId] ++ ids, 2)
+      assert(chunks == ShortChannelIdsChunk(1000, 2, List(ids(0), ids(1))) :: ShortChannelIdsChunk(1002, 2, List(ids(2), ids(3))) :: ShortChannelIdsChunk(1004, 2, List(ids(4), ids(5))) :: Nil)
+    }
+
+    // all ids in the same block
```

Comment:
How about "all ids in same block, chunk size == 2"? This PR creates this very corner case we might as well test it?

Reply:
New extensive tests have been added